src/metaopt/io/parsing.py

```diff
@@ -0,0 +1,257 @@
+             "(default: %s)" % DEF_CMD_MAX_TRIALS[1])
+
+    moptgroup.add_argument(
+        "--pool-size", type=int, metavar='#',
+        help="number of concurrent workers to evaluate candidate samples "
+             "(default: %s)" % DEF_CMD_POOL_SIZE[1])
+
+    moptgroup.add_argument(
+        '-mc', '--moptconfig',
+        type=argparse.FileType('r'), metavar='path-to-config',
+        help="user provided mopt configuration file")
+
+    usergroup = parser.add_argument_group(
+        "User script related arguments",
+        description="These arguments determine user's script behaviour "
+                    "and they can serve as metaopt's parameter declaration.")
+
+    usergroup.add_argument(
+        '-c', '--userconfig',
+        type=str, metavar='path-to-config',
+        help="your script's configuration file (yaml, json, ini, ...anything)")
+
+    usergroup.add_argument(
+        'userscript', type=str, metavar='path-to-script',
+        help="your experiment's script")
+
+    usergroup.add_argument(
+        'userargs', nargs=argparse.REMAINDER, metavar='...',
+        help="command line arguments to your script (if any)")
+
+    args = vars(parser.parse_args())  # convert to dict
+    # Explicitly add metaopt's version as experiment's metadata
+    args['version'] = metaopt.__version__
+    moptfile = args.pop('moptconfig')
+    config = dict()
+    if moptfile:
+        config = yaml.safe_load(moptfile)
+
+    if args['verbose'] == 1:
+        logging.basicConfig(level=logging.INFO)
+    elif args['verbose'] == 2:
+        logging.basicConfig(level=logging.DEBUG)
+
+    return args, config
+
+
+def default_options(user, starttime):
+    """
+    Create a nesteddict with options from the default configuration
+    files, respecting precedence from application's default, to system's and
+    user's.
+
+    .. seealso:: `DEF_CONFIG_FILES_PATHS`
+
+    """
+    defcfg = nesteddict()
+
+    # get some defaults
+    defcfg['exp_name'] = DEF_CMD_EXP_NAME[1].format(
+        user=user,
+        starttime=starttime.isoformat())
+    defcfg['max_trials'] = DEF_CMD_MAX_TRIALS[0]
+    defcfg['pool_size'] = DEF_CMD_POOL_SIZE[0]
+
+    # get default options for some managerial variables (see `ENV_VARS`)
+    for signif, evars in six.iteritems(ENV_VARS):
+        for _, key, default_value in evars:
+            defcfg[signif][key] = default_value
+
+    for cfgpath in DEF_CONFIG_FILES_PATHS:
+        try:
+            with open(cfgpath) as f:
+                cfg = yaml.safe_load(f)
+                if cfg is None:
+                    continue
+                # implies that yaml must be in dict form
+                for k, v in six.iteritems(cfg):
+                    if k in ENV_VARS:
+                        for vk, vv in six.iteritems(v):
+                            defcfg[k][vk] = vv
+                    else:
+                        defcfg[k] = v
+        except IOError as e:  # default file could not be found
```

Comment:
Do we really want to log this as debug(e) and continue on?

Reply:
Not really, I put it to check things out

Reply:
By not really, you mean we don't really want to log.debug(e)? But we want the process to go on? 

Oh ok I think I get it. That means the default value is not defined in the global configuration but we won't complain unless it is not defined in the experiment configuration. Is that it? Then yeah log.debug(e) is good to have. I was confused because I though that was a major exception.